extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -741,6 +741,40 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
     return 0;
   }
 
+  getVolumeDataForSomeReferencedDisplaySet({ viewportData, displaySetService }) {
+    for (const [index, data] of viewportData.data.entries()) {
+      const { imageIds, displaySetInstanceUID } = data;
+
+      const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
+      const referencedDisplaySetInstanceUID = displaySet.measurements?.[0]?.displaySetInstanceUID;
+
+      if (!referencedDisplaySetInstanceUID) {
+        continue;
+      }
+
+      const referencedDisplaySet = displaySetService.getDisplaySetByUID(
+        referencedDisplaySetInstanceUID
+      );
+
+      if (!referencedDisplaySet) {
+        continue;
+      }
+
+      const volume = csToolsUtils.getOrCreateImageVolume(
```

Comment:
I see this is uses twice, maybe we refactor both into using this function 

 if (!volume && displaySet.images) {
        volume = csToolsUtils.getOrCreateImageVolume(displaySet.images.map(image => image.imageId));
      }